### PR TITLE
[MM-54796] Convert `./components/widgets/icons/fa_warning_icon.tsx` to function component

### PR DESCRIPTION
**Pull request by fyfirman**
#### Summary
Convert `./components/widgets/icons/fa_warning_icon.tsx` to function component. I also update the snapshot since it has different result.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/24760
Jira [MM-54796](https://mattermost.atlassian.net/browse/MM-54796)

#### Release Note

```release-note
NONE
```

[MM-54796]: https://mattermost.atlassian.net/browse/MM-54796?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx [hunk 1]
```diff
@@ -8,21 +8,17 @@ import LocalizedIcon from 'components/localized_icon';
...
-        const className = 'fa fa-warning' + (this.props.additionalClassName ? ' ' + this.props.additionalClassName : '');
-        return (
-            <LocalizedIcon
-                className={className}
-                title={{id: t('generic_icons.warning'), defaultMessage: 'Warning Icon'}}
-            />
-        );
-    }
+function WarningIcon(props: Props) {
+    const className = 'fa fa-warning' + (props.additionalClassName ? ' ' + props.additionalClassName : '');
```
Comment (larkox) [inline webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx:15]: I think we can use here `classNames` as `classNames('fa fa-warning', props.additionalClassName)` and should do the same. Do you mind trying?

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx [hunk 2]
```diff
@@ -8,21 +8,17 @@ import LocalizedIcon from 'components/localized_icon';
...
+    return (
+        <LocalizedIcon
+            className={className}
+            title={{id: t('generic_icons.warning'), defaultMessage: 'Warning Icon'}}
```
Comment (larkox) [inline webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx:19]: Out of the scope but could you move this out of the component and use `defineMessage`? That should make the life easier to the linter automation. The idea would be to have something like:
```
const iconTitle = {id: 'generic_icons.warning', defaultMessage: 'Warning Icon'};

function WarningIcon(props:Props) {
    ...
    return (
        ...
            title={iconTitle}
```

That way we can get rid of the `t` function, that is something we want to achieve eventually.

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx [hunk 3]
```diff
@@ -8,21 +8,17 @@ import LocalizedIcon from 'components/localized_icon';
...
-    public render(): JSX.Element {
-        const className = 'fa fa-warning' + (this.props.additionalClassName ? ' ' + this.props.additionalClassName : '');
-        return (
-            <LocalizedIcon
-                className={className}
-                title={{id: t('generic_icons.warning'), defaultMessage: 'Warning Icon'}}
-            />
-        );
-    }
+function WarningIcon(props: Props) {
```
Comment (larkox) [inline webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx:14]: Do you mind making this a arrow function? We want to keep the code as consistent as possible, and we prefer to use arrow functions.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! A few comments a bit out of the scope of the ticket, but I hope you don't mind looking into those :sweat_smile:

**Comment by fyfirman** [comment]
Thanks @larkox for your feedback. It seems pretty clear to me. Let me adjust it along with the other PRs.

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx (hunk 1 above)
Comment (fyfirman) [reply webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx]: I've changed it to `classNames` and I a little bit changes the style to destruct the props like:
```
const WarningProps = ({additionalClassName}:Props) => {
...
}
```
It's more common for me and looking `webapp/channels/src/components/copy_text.tsx` I guess it's convention here.

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx (hunk 2 above)
Comment (fyfirman) [reply webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx]: Just replace it with `defineMessage`. I also confirm it's still working as usual:
image

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx (hunk 3 above)
Comment (fyfirman) [reply webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx]: ✅ Fixed it. I also prefer the arrow function, but looking at other icons which already in function component form like `webapp/channels/src/components/widgets/icons/back_icon.tsx` it use regular functions makes me think twice.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by larkox — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @fyfirman - looks good to me 👍 

cc @larkox

**Comment by larkox** [comment]
@fyfirman There seem to be merge conflicts. Do you mind looking into them?

**Comment by 0xJamin** [comment]
@fyfirman  any updates about this?

@larkox  is it okay if I create a new PR for this since it's been over a month and I can't resolve the conflicts here

**Comment by larkox** [comment]
@0xJamin Yes, please. Feel free to create a new PR.

**Comment by hanzei** [comment]
Replaced by https://github.com/mattermost/mattermost/pull/25644.

Still, thanks for the contribution @fyfirman :+1:

**Pull request closed without merging**